Sources/Sentry/SentryProfiler.mm

```diff
@@ -435,12 +440,30 @@ - (void)start
                     sample[@"queue_address"] = queueAddress;
                 }
 
-                const auto stackIndex = [stacks indexOfObject:stack];
+                const auto stackIndex = [stacks
+                    indexOfObjectPassingTest:^BOOL(NSMutableArray<NSNumber *> *_Nonnull nextStack,
+                        NSUInteger nextStackIdx, BOOL *_Nonnull stopStackEnumeration) {
+                        __block BOOL found = YES;
+                        [nextStack enumerateObjectsUsingBlock:^(NSNumber *_Nonnull nextStackFrame,
+                            NSUInteger nextStackFrameIdx, BOOL *_Nonnull stopFrameEnumeration) {
+                            if (![nextStackFrame isEqualToNumber:stack[nextStackFrameIdx]]) {
+                                *stopFrameEnumeration = YES;
+                                found = NO;
+                            }
+                        }];
+                        if (found) {
+                            *stopStackEnumeration = YES;
+                        }
+                        return found;
+                    }];
```

Comment:
This is an O(n^2) loop where n can be quite big so I would expect this to have some impact on overhead. Does this function get called on the main thread and do we know how long it takes for a larger (e.g. 10 seconds) profile?

Reply:
I converted this to a fast lookup using another NSDictionary. Thanks for catching this!